debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/walkers/LocalVariableTableIterator.java

```diff
@@ -139,16 +142,21 @@ public LocalVariableTable next() {
 				} else {
 					return null;
 				}
+
+				nameSrp = SelfRelativePointer.cast(localVariableTablePtr);
 				name = J9UTF8Pointer.cast(SelfRelativePointer.cast(localVariableTablePtr).get());
 				localVariableTablePtr = localVariableTablePtr.add(SelfRelativePointer.SIZEOF);
-				
+
+				signatureSrp = SelfRelativePointer.cast(localVariableTablePtr);
 				signature = J9UTF8Pointer.cast(SelfRelativePointer.cast(localVariableTablePtr).get());
 				localVariableTablePtr = localVariableTablePtr.add(SelfRelativePointer.SIZEOF);
```

Comment:
These address calculations are wrong for older core files (e.g. there's no `nameSrp` field to skip past).
This will require an update to `VM_LOCAL_VARIABLE_TABLE_VERSION` so we can interpret new and old core files correctly: #11330 should not have been merged without that.

Reply:
#11330 did not change the format of the local variable table. It only changed the `J9VariableInfoValues` struct that is used in the iterator API in the VM. As far as I can tell, DDR does not read this struct anywhere. This PR only changes the way the local variable table is iterated (mirroring the changes in #11330), but it assumes the same format of the underlying data.

Reply:
Ok, I take back my comment about bad address calculations.

What is the point of the extra interpretations of the existing fields?
Why does `LocalVariableTable` need two copies each of 'name', 'signature' and 'genericSignature'?
If the old behavior is still acceptable for `J9BCUtil`, why does `RomClassWalker` need the new interpretations?
Why the duplication in #11330?

Reply:
The reason for the change in #11330 was the need to visit the SRPs to strings in method debug info rather than the strings that they point to, in order to update these SRPs in the serialized ROMClass (serialization basically un-interns all the strings that a ROMClass refers to). The only purpose of this PR is to make the two implementations (in the VM and in DDR) consistent, as was suggested in #11330 reviews.